jetty-unixsocket/src/main/java/org/eclipse/jetty/unixsocket/client/HttpClientTransportOverUnixSockets.java

```diff
@@ -0,0 +1,268 @@
+import org.eclipse.jetty.client.Origin;
+import org.eclipse.jetty.client.SendFailure;
+import org.eclipse.jetty.client.api.Connection;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.client.http.HttpChannelOverHTTP;
+import org.eclipse.jetty.client.http.HttpClientTransportOverHTTP;
+import org.eclipse.jetty.client.http.HttpConnectionOverHTTP;
+import org.eclipse.jetty.client.http.HttpDestinationOverHTTP;
+import org.eclipse.jetty.client.http.HttpSenderOverHTTP;
+import org.eclipse.jetty.io.EndPoint;
+import org.eclipse.jetty.unixsocket.UnixSocketEndPoint;
+import org.eclipse.jetty.util.Promise;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.InetSocketAddress;
+import java.nio.CharBuffer;
+import java.nio.channels.Channels;
+import java.util.List;
+import java.util.Map;
+
+public class HttpClientTransportOverUnixSockets
+    extends HttpClientTransportOverHTTP
+{
+    private String _unixSocket;
+
+    private static final Logger LOGGER = Log.getLogger( HttpClientTransportOverUnixSockets.class );
+
+    public HttpClientTransportOverUnixSockets( String unixSocket )
+    {
+        if ( unixSocket == null )
+        {
+            throw new IllegalArgumentException( "Unix socket file cannot be null" );
+        }
+        this._unixSocket = unixSocket;
+    }
+
+    @Override
+    protected void doStart()
+        throws Exception
+    {
+        super.doStart();
+    }
+
+    @Override
+    protected HttpConnectionOverHTTP newHttpConnection( EndPoint endPoint, HttpDestination destination,
+                                                        Promise<Connection> promise )
+    {
+        return super.newHttpConnection( endPoint, destination, promise );
+    }
+
+    @Override
+    public void connect( InetSocketAddress address, Map<String, Object> context )
+    {
+        // no op
+    }
+
+
+    @Override
+    public HttpDestination newHttpDestination( Origin origin )
+    {
+        return new HttpDestinationOverUnixSocket( getHttpClient(), origin );
+    }
+
+
+
+    class HttpDestinationOverUnixSocket
+        extends HttpDestinationOverHTTP
+    {
+
+        public HttpDestinationOverUnixSocket( HttpClient client, Origin origin )
+        {
+            super( client, origin );
+        }
+
+        @Override
+        protected void send( HttpRequest request, List<Response.ResponseListener> listeners )
+        {
+            super.send( request, listeners );
+        }
+
+        @Override
+        public void send()
+        {
+            if ( getHttpExchanges().isEmpty() )
+            {
+                return;
+            }
+
+            final HttpExchange exchange = getHttpExchanges().poll();
+
+            try
+            {
+                UnixSocketAddress address =
+                    new UnixSocketAddress( HttpClientTransportOverUnixSockets.this._unixSocket );
```

Comment:
Ah I think this is the problem.  `send()` is the wrong extension point.
I think we just need to extend `newSelectorManager`, so we can replace the `newEndPoint` method.
This will avoid the need to rework the send mechanism.